### DISCUSSION TITLE
Unable to add DNS record on Vercel - Vercel and Cloudflare

### DISCUSSION THREAD

**Discussion by ayebhatti** [vercel/community#4537] [Help] [answered]
### Summary

I am using the hobby plan to deploy a site using Vercel, however I would like to host using Cloudflare. I have looked at the Vercel docs to try and configure the DNS as I am getting a verification error. There is no option for me to add any DNS record on Vercel to try and verify domain access. Is that something only available to different plans? If not, how do I go about verifying? I attach screenshots of the Vercel project settings. 

![Screenshot 2023-10-19 at 13 43 18](https://github.com/vercel/community/assets/128622798/a8c9909d-4403-46f8-9423-964d7c2f92c3)
![Screenshot 2023-10-19 at 13 43 27](https://github.com/vercel/community/assets/128622798/d3ee5792-a280-4d4b-80c9-9b418a512750)

### Steps to Reproduce

1. Deploy using Vercel
2. Add domain to a Cloudflare account
3. Attempt to add Cloudflare nameserver's to Vercel project

**Comment by amyegan** [comment]
Hi @ayebhatti. If you're trying to use an external domain with a Vercel account, you should [add the custom domain to your project settings](https://vercel.com/docs/projects/domains/add-a-domain#) rather than adding nameservers. You can point the domain to Vercel by adding or modifying DNS records and settings in your Cloudflare account.

https://vercel.com/docs/integrations/cloudflare

**Comment by BeeBooBap** [comment]
Hi @amyegan thanks for getting back to me. 

I followed the steps on the link provided however the integration is still not complete as Cloudflare is waiting for the nameserver update on Vercel. 

![Screenshot 2023-10-20 at 09 29 54](https://github.com/vercel/community/assets/72410789/abf0534d-10ac-454b-8941-a10f9fa141a1)

I added both nameserver's as domains to the project however I then get a verification issue. 

![Screenshot 2023-10-20 at 09 31 27](https://github.com/vercel/community/assets/72410789/bd969ec8-78e7-497e-a9cf-79a8f9476161)

Do you know how I can go about verifying the domain I added to Vercel by setting the TXT record on _vercel.cloudflare.com? I tried following this page to [manage DNS records](https://vercel.com/docs/projects/domains/managing-dns-records) however I dont seem to have an advanced settings option for the domain to update DNS records. 

Any help on this would be great! Thanks.

**Reply by amyegan** [reply, marked as answer]
The nameservers for `vercel.app` are set for use with Vercel projects. You cannot move your `*.vercel.app` domain to Cloudflare. When hosting an app with Cloudflare Pages, they will provide their own `*.pages.dev` domain for your project.

**Comment by andrewmartin** [comment]
@amyegan 

I am having a very similar loop here.

What I want to do is move over to `ns1.vercel-dns.com` and `ns2.vercel-dns.com` from my existing nameservers `ns1.digitalocean.com` etc). I have the site already deployed and working on Vercel.com with this domain, but I just want to kill off my DO server and settings so I don't need to keep that account managing my DNS.

1. I had my stuff working fine (on Vercel) when using some (old) Digital Ocean nameservers (ns1.digitalocean.com).
2. I tried swapping to `ns1.vercel-dns.com` and `ns2.vercel-dns.com` at my registrar.
3. Whenever I do that, I end up in this crazy loop where I cannot ever see the button in this screenshot:

![Adding   Configuring a Custom Domain 2024-05-04 at 8 06 21 PM](https://github.com/vercel/community/assets/273842/0d0d448e-bac8-4ac5-9ab0-42172c8dfb44)

4. I end up seeing some crazy verification text and I cannot get around it:
![releasewave - Overview – Vercel 2024-05-04 at 8 06 52 PM](https://github.com/vercel/community/assets/273842/3c7adb14-2e43-4d8d-9b3f-884eb597b303)

Am I doing something wrong here?
